### ISSUE TITLE
Downloads button: Text colour issue (Tycho)

### ISSUE THREAD

**Issue by THEtomaso** [MoonchildProductions/Pale-Moon#695]
The text for the time remaining counter, on the Downloads button in the Navigation Toolbar, doesn't detect theme colors.
With my current settings, the text is almost impossible to make out!

(Pale Moon v27.0.0b3,x64 / Windows 8.1 Pro / ColorizationColorBalance = 0)

**Comment by THEtomaso** [comment]
I think this problem needs to be fixed with the same code that were used for the recent Menu Bar issue:
https://forum.palemoon.org/viewtopic.php?f=3&t=12311/

**Comment by janekptacijarabaci** [comment]
It's there for a long time (i.e. PM 25/26).

@Lootyhoof 
AFAIK: Perhaps it would be worth to think about it:
https://github.com/MoonchildProductions/Tycho/commit/f5a6892f6e9ef93296f77a46955c86b86bfd7195#diff-aecd6d0813d0cb62fd8e3d334b5a5503R405

**Comment by janekptacijarabaci** [comment]
After #845 (white on light background, black on dark background):

~~Windows 7 - aero - transparency: enable~~
~~Windows 7 - aero - transparency: disable~~
~~Windows 7 - basic~~
~~Windows 7 - classic~~
~~Windows 7 - high contrast 1~~
~~Windows 7 - high contrast 2~~
~~Windows 7 - high contrast - black~~
~~Windows 7 - high contrast - white~~

~~Windows 8 - default~~
~~Windows 8 - basic theme~~
Windows 8 - high contrast 1
~~Windows 8 - high contrast 2~~
Windows 8 - high contrast - black
Windows 8 - high contrast - white

Windows 10 Redstone 3+? - aero - transparency: enable/disable ?
Windows 10 - basic - show colors on title bar: on
~~Windows 10 - basic - show colors on title bar: off~~
Windows 10 - high contrast 1
~~Windows 10 - high contrast 2~~
Windows 10 - high contrast - black
Windows 10 - high contrast - white

**Comment by wolfbeast** [comment]
I guess to completely fix this, it needs a `[brighttext]` selector on `#downloads-indicator-counter:not(:-moz-lwtheme)`?

**Comment by Lootyhoof** [comment]
>I guess to completely fix this, it needs a [brighttext] selector on #downloads-indicator-counter:not(:-moz-lwtheme)?

That won't do anything that isn't already being done. `[brighttext]` being used at all includes both lwtheme and non-lwtheme modes.

The issue is mostly with high contrast mode, a mode which we've had issue with in the past as we can't detect when we're using it. This normally wouldn't be an issue, due to `[brighttext]`, however `[brighttext]` doesn't seem to want to detect bright text in high contrast mode.

Additionally @janekptacijarabaci I can't seem to confirm it not working on Windows 10 with coloured titlebars. I take it you restart the browser each time you change the window metrics? It would also seem you need to switch between tabs on top and tabs not on top to make it switch between each if using a different shade of colour previously.

![Dark text](http://i.imgur.com/AqrGDKz.png) ![Light text](http://i.imgur.com/3Vq7C9F.png)

**Comment by janekptacijarabaci** [comment]
> I take it you restart the browser each time you change the window metrics?

Yes. But see also:
https://github.com/MoonchildProductions/Pale-Moon/blob/27.0.0b3r2/toolkit/mozapps/extensions/LightweightThemeManager.jsm#L699
vs.
https://github.com/MoonchildProductions/Pale-Moon/pull/437/files#diff-cef53509552274ae809ff183f3394680R7214

Windows 10 - basic - show colors on title bar: on
![win10_1](https://cloud.githubusercontent.com/assets/2373486/22126092/81f8be28-de96-11e6-9b9c-3165e3d91a57.png)

Windows 10 - high contrast 1
![win10_2](https://cloud.githubusercontent.com/assets/2373486/22126106/8da679b8-de96-11e6-80cc-c666e7743dfa.png)

Windows 10 - high contrast - black
![win10_3](https://cloud.githubusercontent.com/assets/2373486/22126110/913c7690-de96-11e6-82ae-b8b801d422c5.png)

Windows 10 - high contrast - white
![win10_4](https://cloud.githubusercontent.com/assets/2373486/22126124/9b1a1e56-de96-11e6-9258-7ab11991f096.png)

----------

IMHO: I have no special settings...

**Comment by wolfbeast** [comment]
> IMHO: I have no special settings...

... Apart from using high-contrast which is a special accessibility mode that you should normally never use.

As for the straight-blue: that's doing things correctly. If you look at the caption controls, those are white, so the text on the button should be white as well. Unfortunately it's not the easiest on the eyes but that's just because of the blue (which has a low contrast-contributing value) in combination with it being small antialiased text on a gradient button that lightens what the text is on.

Short of removing the gradient for bright text situations, I don't see that much can be done. Unless you want to outline the text, but that will likely get ugly as well.

**Comment by wolfbeast** [comment]
Closing this issue. improvements of the gradient of toolbar buttons in brighttext situations should be handled as a separate issue to this.
